File: dotnet/external-user-service/UserService/shala.api/startup/configurations/app.extensions/AppSeederExtensions.cs
namespace shala.api.startup.configurations;

public static class AppSeederExtensions
{
    public static WebApplication UseSeeder(this WebApplication app)
    {
        //Get service provider from app
        var serviceProvider = app.Services;

        var isDevEnvironment = app.Environment.IsDevelopment();
        var isTestEnvironment = app.Environment.IsEnvironment("Test");

        Seeder.SeedAsync(serviceProvider, isDevEnvironment, isTestEnvironment).Wait();
        return app;
    }
}

File: python/python_b2b_seeding/python_service/startup/configurations/app.extensions/app_seeder_extensions.py
"""
Application seeder extensions.
Similar to the dotnet AppSeederExtensions.cs implementation.
This provides a helper function to use the seeder, matching the dotnet extension pattern.
"""
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from fastapi import FastAPI


def use_seeder(app: 'FastAPI'):
    """
    Extension method to use seeder on application startup.
    Similar to dotnet UseSeeder extension method.
    
    Args:
        app: FastAPI application instance
        
    Returns:
        The app instance (for method chaining)
    """
    from config.config import Config
    from startup.seeder import Seeder
    
    # Get environment info
    is_dev_environment = Config.is_development()
    is_test_environment = Config.is_test()
    
    # Run seeding (synchronously as dotnet does)
    if Config.AUTO_SEED_ON_STARTUP:
        Seeder.seed_async(is_dev_environment, is_test_environment)
    
    return app

